### PR TITLE
refactor(core): Port `generic` config

### PR DESCRIPTION
**Pull request by ivov**
Follow-up to #10321

### CODE REVIEW

File: packages/@n8n/config/src/configs/generic.config.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { Config, Env } from '../decorators';
+
+@Config
+export class GenericConfig {
+	/** Default timezone for the n8n instance. Can be overridden on a per-workflow basis. */
+	@Env('GENERIC_TIMEZONE')
+	timezone: string = 'America/New_York';
+
+	@Env('N8N_RELEASE_TYPE')
```
Comment (netroy) [inline packages/@n8n/config/src/configs/generic.config.ts:9]: Do we really want to keep the `geenric` section? or should we start moving these to more relevant sections, like `release.type`, and `system.gracefulShutdown.timeout` ?

File: packages/cli/src/config/index.ts [hunk 2]
```diff
@@ -122,7 +122,7 @@ if (executionProcess === 'own') {
 }
 
 setGlobalState({
-	defaultTimezone: config.getEnv('generic.timezone'),
+	defaultTimezone: Container.get(GlobalConfig).generic.timezone,
```
Comment (netroy) [inline packages/cli/src/config/index.ts:125]: At some point we should also stop using `GlobalConfig` everywhere. Every config class is already individually injectable.

**Review by netroy — APPROVED** [review]

File: packages/@n8n/config/src/configs/generic.config.ts (hunk 1 above)
Comment (ivov) [reply packages/@n8n/config/src/configs/generic.config.ts:9]: Agree, eventually let's reorganize this and some of the others to make more sense. For now focusing on bringing them all into the new style as they are now, also minimizes errors when porting.

**Comment by janober** [comment]
Got released with [`n8n@1.65.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.65.0)

**Pull request merged**